config/config.go

```diff
@@ -83,8 +89,22 @@ func Get() *SourcesApiConfig {
 
 		// Grab the first broker
 		kafkaBroker := cfg.Kafka.Brokers[0]
-
 		options.SetDefault("KafkaBrokers", []string{fmt.Sprintf("%s:%v", kafkaBroker.Hostname, *kafkaBroker.Port)})
+
+		if kafkaBroker.Authtype != nil {
+			options.SetDefault("KafkaSaslEnabled", true)
+
+			caPath, err := cfg.KafkaCa(kafkaBroker)
+			if err != nil {
+				log.Fatalf(`Could not write the Kafka broker's CA to a temporary directory: %s`, err)
+			}
+
+			options.SetDefault("KafkaSaslCaPath", caPath)
```

Comment:
can we test those cars for emptiness/nil ? + log them in case of when there are empty/nil - it could help with potential debugging.

Reply:
Sure, I've added those nil checks and logged the values except for the sensitive credentials. I was doubting if I should simply `Fatal` if any of the `mechanism`, `username` or `password` values are empty, since without those we wouldn't be able to connect to Kafka... Everything would work, but not Kafka.